File: leetcode/C++/is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int idx = 0;
        for(int i=0; i<t.length()&&idx<s.length(); ++i) if(s[idx] == t[i]) ++idx;
        return idx==(int)s.length();
    }
};

File: leetcode/Python/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        idx = 0
        if len(s) == 0:
            return True
        for i in range(len(t)):
            if s[idx] == t[i]:
                idx += 1
            if idx == len(s):
                break
        return True if idx == len(s) else False